### PR TITLE
Inherit from CharField instead of Field

### LINKED ISSUE #445: Field of CharField inheritance
Most of localflavor form fields inherit from `CharField`. However, some are directly inheriting from `Field`. Whenever you use those fields as the default form field for any model CharField, you'll get the `__init__() got an unexpected keyword argument 'max_length'` TypeError. I wonder if the fact some fields inheriting from `Field` only is a desing decision or simply an oversight.

@benkonrath, what do you think?

### PR DESCRIPTION
**Pull request by benkonrath**
A first (incomplete) PR for converting the fields inheriting from `Field` to inheriting from `CharField`.

Fixes  #445

### CODE REVIEW

File: localflavor/at/forms.py [hunk 1]
```diff
@@ -53,8 +52,8 @@ class ATSocialSecurityNumberField(Field):
 
     def clean(self, value):
         value = super().clean(value)
-        if value in EMPTY_VALUES:
-            return ""
+        if value in self.empty_values:
+            return self.empty_value
```
Comment (claudep) [inline localflavor/at/forms.py:56]: Could/should those empty_values changes be part of a separate commit?

**Comment by claudep** [comment]
I would also make a separate PR for isort changes.

**Comment by benkonrath** [comment]
The isort changes are mostly relevant but I get your point - reverted.

File: localflavor/at/forms.py (hunk 1 above)
Comment (benkonrath) [reply localflavor/at/forms.py:56]: The reason I included these changes was because changing to `CharField` would introduce the option to set `empty_value` as an argument without it having any effect. `self.empty_value` defaults to an empty string and `self.empty_values` defaults to `EMPTY_VALUES` so there shouldn't be a problem with this change.

Does this reasoning sound ok to you? I'm definitely open to less invasive changes if you think that's better for a 3.1.1 release and then following up with the empty value changes in 4.0.

Comment (claudep) [reply localflavor/at/forms.py:56]: Less hard-coding sounds fine!

Comment (claudep) [reply localflavor/at/forms.py:55]: In fact, I would simply return `value`, as the conversion of the value to self.empty_value is already happening in `to_python`. Even if the result should be exactly the same.

Comment (benkonrath) [reply localflavor/at/forms.py:55]: Makes sense. I've been using the `return self.empty_value` pattern for a while and I think it should be changed everywhere because existing fields are often used a reference when new fields are added. This can happen in another PR.

**Comment by claudep** [comment]
The more I think about it, the less I like those `strip` parts. As you mentions, the default is `True` anyway, so I don't see the point in twiddling with that. I tried your PR without those parts in #462 and tests seem to pass (at least locally).

**Comment by benkonrath** [comment]
That's fine. Feel free to merge your modified version.
